docs/snapshotting/snapshot-support.md

```diff
@@ -0,0 +1,280 @@
+of older versions that we can restore from / save a snapshot to, from the current
+version) will be defined later.
+
+## Snapshot API
+
+Firecracker exposes the following APIs for manipulating snapshots: `Pause`, `Resume`
+and `CreateSnapshot` can be called only after booting the microVM, while `LoadSnapshot`
+is allowed only before boot.
+
+### Pausing the microVM
+
+To create a snapshot, you have to firstly pause the running microVM and its vCPUs with
+the following API command:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PATCH "http://localhost/vm" \
+    -H "accept: application/json" \
+    -H "Content-Type: application/json" \
+    -d "{
+            "state": "Paused"
+    }"
+```
+
+**Prerequisites**: Pausing the microVM can be done any time after the microVM is booted.
+                   Successive calls of this request keep the microVM in the `Paused`
+                   state.
+**Effects**:
+- _on success_: microVM is guaranteed to be `Paused`.
+- _on failure_: no side-effects.
+
+*Note*: If you change your mind and want to create a snapshot later, you can resume the
+        microVM and then pause it again when you decide to take that snapshot.
+
+## Creating snapshots
+
+Now that the microVM is paused, you can create a snapshot, which can be either a `full`
+one or a `diff` one. Full snapshots always create a complete, resume-able snapshot of
+the current microVM state and memory. Diff snapshots save the current microVM state
+and the memory dirtied since the last snapshot (full or diff). Diff snapshots are not
+resume-able, but can be merged into a full snapshot using external (provided) tooling.
+
+### Creating full snapshots
+
+For creating a full snapshot, you can use the following API command:
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PUT "http://localhost/snapshot/create" \
+    -H  "accept: application/json" \
+    -H  "Content-Type: application/json" \
+    -d "{
+            \"snapshot_type\": \"Full\",
+            \"snapshot_path\": \"/path/to/snapshot_file\",
+	        \"mem_file_path\": \"/path/to/mem_file\",
+            \"version\": \"2\"
+    }"
+```
+
+Details about the required and optional fields can be found in the
+[swagger definition](../../src/api_server/swagger/firecracker.yaml).
+
+**Prerequisites**: It can be called only when the microVM is `Paused`.
+**Effects**:
+- _on success_:
+  - The file indicated by `/path/to/snapshot_file` contains the devices' model state
+    and emulation state. The one indicated by `/path/to/mem_file` contains a full copy
+    of the guest memory.
+  - The block devices contents are flushed to their backing files. At this point, these
+    should be backed up externally by the user. The generated snapshot files are
+    immediately available to be used (current process releases ownership).
+  - The snapshot creation resets then the dirtied page bitmap and marks all pages clean
```

Comment:
This should happen only if diff snapshots are enabled when booting/restoring.

Reply:
Updated it.